### ISSUE TITLE
Support dim order in Java Tensor

### ISSUE THREAD

**Issue by GregoryComer** [pytorch/executorch#11593]
### 🚀 The feature, motivation and pitch

Our C++ tensor APIs support dim order, allowing for elements in memory to be laid out in an alternative order. This is commonly used to support channels-last image tensors, where channel values can be interleaved, rather than being laid out in the default channels first order. This would be useful to implement on Java tensors.

## Goals for this task:
 * Add a dim order field to [Tensor.java](https://github.com/pytorch/executorch/blob/main/extension/android/executorch_android/src/main/java/org/pytorch/executorch/Tensor.java).
   * Dim order is represented as a list of integers, similar to sizes. It should look very similar to the [shape accessor](https://github.com/pytorch/executorch/blob/main/extension/android/executorch_android/src/main/java/org/pytorch/executorch/Tensor.java#L367).
 * Add a constructor to Tensor.java taking dim order. Update the various [fromBlob](https://github.com/pytorch/executorch/blob/main/extension/android/executorch_android/src/main/java/org/pytorch/executorch/Tensor.java#L146) methods to add an overload taking dim order.
   * See [https://github.com/pytorch/executorch/blob/main/extension/tensor/tensor_ptr.h#L284] for an example of how this is done in the native tensor builder.
   * See the existing constructor here: [Tensor.java](https://github.com/pytorch/executorch/blob/main/extension/android/executorch_android/src/main/java/org/pytorch/executorch/Tensor.java#L346).
 * Update [newJTensorFromTensor](https://github.com/pytorch/executorch/blob/main/extension/android/jni/jni_layer.cpp#L54C3-L54C23) to support tensors with non-default dim orders.
   * When a native tensor with non-default dim order is passed in, it should create a Java tensor with the same dim order.
 * Update [JEValueToTensorImpl](https://github.com/pytorch/executorch/blob/main/extension/android/jni/jni_layer.cpp#L155) to support non-default dim orders.
   * When a Java tensor with non-default dim order is passed in, it should construct a native tensor with the same dim order.

## Testing the change
 * We'll want to add a new test to [ModuleInstrumentationTest.kt](https://github.com/pytorch/executorch/blob/main/extension/android/executorch_android/src/androidTest/java/org/pytorch/executorch/ModuleInstrumentationTest.kt) to cover inference with a channels last tensor.
   * We'll also need to generate a test model that takes channels-last inputs (see below).
   * This test should create an input tensor in channels-last format (using the newly created methods), run Add (or another model), and check the output values (including dim order). See [TestMV2FP32](https://github.com/pytorch/executorch/blob/db96abad9e79e999d95276b39aa6404842f84d25/extension/android/executorch_android/src/androidTest/java/org/pytorch/executorch/ModuleE2ETest.kt#L77) for a very similar test.

### Generating a channels-last test model
We use a simple model (ModuleAdd.pte) for existing tests in this file. We'll want to create a similar model that is exported with channels last tensors. We'll want to do something similar for this. An example export script might look like this:

```python
import torch
from executorch.exir import to_edge_transform_and_lower

class ModuleAddChannelsLast(torch.nn.Module):
  def forward(self, x, y):
    return x + y

inputs = (torch.randn(1, 3, 16, 16).to(memory_format=torch.channels_last), torch.randn(1, 3, 16, 16).to(memory_format=torch.channels_last))
ep = torch.export.export(ModuleAddChannelsLast(), inputs)
lowered = to_edge_transform_and_lower(ep).to_executorch()
with open("ModuleAddChannelsLast.pte", "wb") as f:
   f.write(lowered.buffer)
```

The existing export code for ModuleAdd.pte is here: [link](https://github.com/pytorch/executorch/blob/db96abad9e79e999d95276b39aa6404842f84d25/extension/runner_util/test/CMakeLists.txt#L22). We'll want to add another model, maybe called ModuleAddChannelsLast.

**Comment by kamalesh0406** [comment]
@GregoryComer can you assign this issue to me? I am working on it.

**Comment by krish-arora-88** [comment]
Hey @kamalesh0406! If you're not working on it anymore, I'd be happy to take it off your hands! Please let me know.

Thanks!
